r-package/inst/include/models/adult_hiv_model_simulation.hpp

```diff
@@ -158,7 +168,108 @@ struct AdultHivModelSimulation<Config> {
     }
   };
 
-  void run_new_p_infections(int hiv_step) {
+  void calc_new_infections_incidmod_transmrate(int hiv_step) {
+    
+    const auto& p_ha = pars.ha;
+    auto& n_ha = state_next.ha;
+    auto& n_dp = state_next.dp;
+    auto& i_ha = intermediate.ha;
+    
+    // sum population sizes
+    real_type Xhivn_s[NS];
+    real_type Xhivn_incagerr[NS];
+    real_type Xhivp_noart = 0.0;
+    real_type Xart = 0.0;
+
+
+    for(int s = 0; s < NS; ++s){
+      Xhivn_s[s] = 0.0;
+      Xhivn_incagerr[s] = 0.0;
+      for(int a = SS::pIDX_15to49; a < SS::pIDX_15to49 + SS::pAG_15to49; ++a) {
+	auto Xhivn_sa = n_dp.p_totpop(a, s) - n_ha.p_hivpop(a, s);
+	Xhivn_s[s] += Xhivn_sa; 
+	Xhivn_incagerr[s] += p_ha.incidence_rate_ratio_age(a - p_ha.pIDX_INCIDPOP, s, t) * Xhivn_sa;
+      }
+
+      for(int ha = SS::hIDX_15to49; ha < SS::hIDX_15to49 + SS::hAG_15to49 + 1; ++ha){
+
+	// adjustment to first and last age group for partial year time step
+	// calculation proportion of HIV population to include / exclude based
+	// on hivpop in single-year ages.
+	real_type prop_include;
+	if(ha == SS::hIDX_15to49){
+	  real_type hivp_ha = 0.0;
+	  int a = SS::pIDX_15to49;
+	  for(int i = 0; i < hAG_span[ha]; ++i, ++a) {
+	    hivp_ha += n_ha.p_hivpop(a, s);
+	  }
+	  prop_include = (hivp_ha > 0) ? 1.0 - n_ha.p_hivpop(SS::pIDX_15to49, s) / hivp_ha * (1.0 - opts.dt * hiv_step) : 1.0;
```

Comment:
I feel like at some point we had decided to use the explicit if/else formatting for these types of things to help with readability